### ISSUE TITLE
[QUESTION] Modify/visualize the data from Linux shell or GUI

### ISSUE THREAD

**Issue by lpares12** [litedb-org/LiteDB#2358]
Hi,

Is it possible to modify or visualize the data from a linux shell or GUI? All I find are references to a LiteDBShell.exe file, but of course that's not useful for linux. And also, I see the GUI is also only for windows.

Would like to know if there's any solution to that!

Thanks.

**Comment by eabase** [comment]
BUMP!

**Comment by JKamsker** [comment]
Yes. LiteDB.Shell is a plain .NET console app (it targets `net8.0` on the `dev` branch), not a Windows-only tool. We just don't ship a prebuilt binary, which is why you only find references to an `.exe`. With the .NET 8 SDK, run it from a clone:

```bash
git clone --branch dev https://github.com/litedb-org/LiteDB.git
cd LiteDB
dotnet run --project LiteDB.Shell -- --exec "open /path/to/mydb.db"
```

Then type `help`, or SQL statements terminated with `;`:

```sql
SELECT $ FROM orders WHERE YEAR(Created) <= 2019;
UPDATE orders SET Status = "Closed" WHERE _id = 65;
```

Open the file with the `open` command as shown. The shell's argument parser treats anything starting with `/` as an option, so passing an absolute Linux path directly (`-- /path/to/mydb.db`) silently opens nothing. A relative path (`-- mydb.db`) works.

It can also be scripted:

```bash
dotnet run --project LiteDB.Shell -- --exec "open /path/to/mydb.db" --exec 'SELECT $ FROM orders;' --exit
dotnet run --project LiteDB.Shell -- --exec "open /path/to/mydb.db" --run script.txt --exit
```

If you only have a newer SDK (9 or 10), prefix the command with `DOTNET_ROLL_FORWARD=Major`. A file created with LiteDB 5.0.21 and edited this way still opens in 5.0.21. Source: [LiteDB.Shell](https://github.com/litedb-org/LiteDB/tree/dev/LiteDB.Shell).

For a GUI: LiteDB.Studio is a WinForms app on .NET Framework, so it runs on Windows only. A community option is [OneBella](https://github.com/namigop/OneBella), which describes itself as a "cross-platform (Windows, MacOS and Linux) GUI tool for viewing and editing LiteDB 5+ databases". It's third-party and not maintained by us.

Closing this as answered — if it doesn't cover your case, comment here and we'll reopen it.

The Shell ignoring absolute Unix paths is tracked in #2774, so I'm closing this one as a duplicate of it. If this doesn't cover your case, comment here and we'll reopen it.
